### PR TITLE
Feat/top lvl blockchain

### PR DESCRIPTION
**Pull request by CharlieC3**
## Description

Moves `Stacks Blockchain` on left navbar to a higher, more noticeable area up top.
Renames other top-level menu items and updates all references.

Closes https://github.com/blockstack/docs/issues/868

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [X] API reference/documentation update
- [ ] Other

## Checklist
- [X] `npm run lint` passes

### CODE REVIEW

**Comment by CharlieC3** [comment]
@aulneau For some reason after renaming the `stacks-blockchain` directory to `understand-stacks-2.0`, clicking on the menu item in the navbar results in an error instead of directing to the `overview` page underneath this item:
Screen Shot 2020-11-06 at 3 47 37 PM

If I try removing the `.` in the name, it resolves correctly. I've tried a few things, including quoting the path to the page and using the HTML encoding for `.` to no avail. Any idea on how this should be fixed?

**Comment by aulneau** [comment]
Yeah, makes sense that a period would break that. Would it be possible to do `understanding-stacks-2` or `2-0`?

File: next.config.js [hunk 1]
```diff
@@ -352,8 +352,16 @@ async function redirects() {
     },
     { source: '/org/terms.html', destination: '/references/glossary', permanent: true },
     // overview redirects
-    { source: '/stacks-blockchain', destination: '/stacks-blockchain/overview', permanent: true },
```
Comment (agraebe) [inline next.config.js:355]: we shouldn't change existing redirect sources. For these particular pages, we should set the destination to the new URL

File: next.config.js [hunk 2]
```diff
@@ -363,23 +371,23 @@ async function redirects() {
     { source: '/references', destination: '/references/blockstack-cli', permanent: true },
     { source: '/ecosystem', destination: '/ecosystem/overview', permanent: true },
     {
-      source: '/stacks-blockchain/testnet-node',
```
Comment (agraebe) [inline next.config.js:366]: same as the last comment - let's update the destination here but leave the source as is

File: src/common/navigation.yaml [hunk 3]
```diff
@@ -45,32 +71,6 @@ sections:
           - title: Tutorials
             pages:
               - path: /integrate-indexing
-      - path: /stacks-blockchain
```
Comment (agraebe) [inline src/common/navigation.yaml:48]: all these "old" URLs need to be matched with new redirects - would double check that

**Comment by agraebe** [comment]
> Yeah, makes sense that a period would break that. Would it be possible to do `understanding-stacks-2` or `2-0`?

alternatively, we could use `understanding-stacks` in the URLs/filenames and specify a title for the page itself to include the `2.0`

**Comment by CharlieC3** [comment]
> > Yeah, makes sense that a period would break that. Would it be possible to do `understanding-stacks-2` or `2-0`?
> 
> alternatively, we could use `understanding-stacks` in the URLs/filenames and specify a title for the page itself to include the `2.0`

@agraebe 
I think this makes sense and it would be more future proof. Originally you wanted `understand` though, are we changing that to `understanding`?

**Comment by agraebe** [comment]
sorry, meant to say `understand`

File: next.config.js (hunk 2 above)
Comment (CharlieC3) [reply next.config.js:366]: I added back these redirects and fixed their destination fields.

**Comment by agraebe** [comment]
@CharlieC3 let's merge this at the end of the docs week - with so many PRs in flight, I'm afraid we'd have to fix every other PR if we make this change now

**Comment by agraebe** [comment]
I think we can pick up this PR again and rebase whenever you get a chance @CharlieC3 :)

**Comment by CharlieC3** [comment]
@agraebe For some reason Vercel is throwing errors when loading certain pages, like the "understand stacks" or "write smart contracts" pages, but they load fine in Gitpods. But, rebase/update is complete. Please review when you get a chance!

**Comment by agraebe** [comment]
> @agraebe For some reason Vercel is throwing errors when loading certain pages, like the "understand stacks" or "write smart contracts" pages, but they load fine in Gitpods. But, rebase/update is complete. Please review when you get a chance!

if I read the Vercel preview comment correctly, it built with commit `5f336c0`, which isn't available. I'm assuming it just simply didn't deploy the latest changes

**Comment by agraebe** [comment]
No, I was wrong. It's something different:

```js
main-15e3749a34e8bd3bf3b0.module.js:1 TypeError: Cannot read property 'path' of undefined
    at _app-7df6f64cf9fb60522cd5.module.js:1
    at Array.map (<anonymous>)
    at y.Y [as constructor] (_app-7df6f64cf9fb60522cd5.module.js:1)
    at y.N [as render] (framework.dba7cce335f14b902108.module.js:1)
    at O (framework.dba7cce335f14b902108.module.js:1)
    at C (framework.dba7cce335f14b902108.module.js:1)
    at A (framework.dba7cce335f14b902108.module.js:1)
    at O (framework.dba7cce335f14b902108.module.js:1)
    at C (framework.dba7cce335f14b902108.module.js:1)
    at O (framework.dba7cce335f14b902108.module.js:1)
```

In fact, my window is freezing once I navigate to the "Understand stacks" page - there might be a redirect loop in the file?

**Comment by CharlieC3** [comment]
@agraebe If you try running this branch in a gitpod, you'll see it works though. In summary, it doesn't work in Vercel, but it does in Gitpods.

**Comment by agraebe** [comment]
You're right! @aulneau - do you have an idea what that might be? There was a rebase and maybe Vercel got hung up on the force push?

**Comment by CharlieC3** [comment]
Also, the Clarity Values page is wonky in what's currently deployed to prod. If you go to the [smart contracts overview page](https://docs.blockstack.org/smart-contracts/overview) and click on the "Clarity values" page in the sidebar, the sidebar goes back to the route, but the clarity values page still renders luckily. However it does _not_ render in Gitpods.

If this is a simple issue I can fix it in this PR, but this was not introduced from this PR, and is a bug live right now.

**Comment by aulneau** [comment]
I see, there is some general weirdness happening across the site. I cannot look at it today, but will spend some time fixing up the docs this week.

**Comment by agraebe** [comment]
@aulneau would love to get that fixed. could you take a look please?

**Comment by aulneau** [comment]
@agraebe this is an issue with ublock origin, because the name is `clarity-values`, it's blocked automatically. we will have to find a different name for this page that is not `clarity-*`

**Comment by agraebe** [comment]
> @agraebe this is an issue with ublock origin, because the name is `clarity-values`, it's blocked automatically. we will have to find a different name for this page that is not `clarity-*`

ah, that makes sense! I just looked at [the page](https://docs.blockstack.org/smart-contracts/clarity-values) and I think we could rename it to `types.md`. The first chapter should also link out to the [types references doc](https://docs.blockstack.org/references/language-types), but that might be for another PR.

@CharlieC3 could you fix and resolve conflicts, so we can go ahead and merge? :)

File: src/common/navigation.yaml [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
-      - path: /build-an-app # is an overview page
-      - path: /smart-contracts
+      - path: /build-apps # is an overview page
+      - path: /write-smart-contracts
         pages:
           - path: /overview
           - path: /principals
-          - path: /clarity-values
+          - path: /cli-wallet-quickstart
+          - path: /install-source
```
Comment (CharlieC3) [inline src/common/navigation.yaml:10]: These two pages were seemingly never linked to after they were written. Not sure if that was intentional, but I added links to them in this change.

**Comment by CharlieC3** [comment]
> @CharlieC3 could you fix and resolve conflicts, so we can go ahead and merge? :)

Change made, was this what you had in mind?
https://blockstack-docs-6ynimxaid.vercel.app/write-smart-contracts/types

File: src/pages/write-smart-contracts/public-registry-tutorial.md [hunk 5]
```diff
@@ -31,14 +31,14 @@ By the end of this tutorial, you will...
...
+Furthermore, the to-do app will interact with a smart contract deployed as `ST1234....todo-registry`. The contract source code is available at [github](https://github.com/friedger/blockstack-todos/blob/tut/step1/contracts/todo-registry.clar). There may already be a deployed version available on the testnet; the [testnet explorer](https://testnet-explorer.blockstack.org/) can be used to search for it. Alternatively, the contract can be deployed as described in the [hello world tutorial](/smart-contracts/hello-world-tutorial#step-5-deploy-the-contract). Then you have to use the corresponding contract address and name in this tutorial. Throughout this tutorial, we use `ST3YPJ6BBCZCMH71TV8BK50YC6QJTWEGCNDFWEQ15.todo-registry` as an example.
```
Comment (CharlieC3) [inline src/pages/write-smart-contracts/public-registry-tutorial.md:34]: Fixed some typos in reference to #910

File: src/common/navigation.yaml (hunk 4 above)
Comment (agraebe) [reply src/common/navigation.yaml:10]: linking to them is probably right. I have an issue open to update the CLI quickstart page as well

Comment (agraebe) [reply src/common/navigation.yaml:10]: ah, wait - the install source page seems unfinished. wouldn't link to it (and maybe remove the file entirely?)

**Comment by agraebe** [comment]
> > @CharlieC3 could you fix and resolve conflicts, so we can go ahead and merge? :)
> 
> Change made, was this what you had in mind?
> https://blockstack-docs-6ynimxaid.vercel.app/write-smart-contracts/types

yeah, I did. looking at it now though, `types` might not be as appropriate as `values.md`. Would fit the metadata of the file and content better. Sorry for the additional request :) 

Could we also move the link in the nav right below the `Principals` link? The wallet should be lower in priority in the navigation 🙏

**Comment by CharlieC3** [comment]
> yeah, I did. looking at it now though, `types` might not be as appropriate as `values.md`. Would fit the metadata of the file and content better. Sorry for the additional request :)
> 
> Could we also move the link in the nav right below the `Principals` link? The wallet should be lower in priority in the navigation 🙏

Done, also removed the link to the `install-source` page, but didn't delete the page itself.

**Comment by CharlieC3** [comment]
@agraebe Any other comments? Good to merge?

**Review by agraebe — APPROVED** [review]

**Pull request merged**